src/instrumentations/https/HttpInstrumentation.ts

```diff
@@ -4,29 +4,43 @@ import { HttpHooks } from './http';
 import { Instrumentor } from '../instrumentor';
 
 export default class LumigoHttpInstrumentation extends Instrumentor<HttpInstrumentation> {
+  private readonly ignoredHostnames: string[];
+
+  constructor(...ignoredHostnames: string[]) {
+    super();
+
+    this.ignoredHostnames = (ignoredHostnames || []).concat(
+      [process.env.ECS_CONTAINER_METADATA_URI, process.env.ECS_CONTAINER_METADATA_URI_V4]
+        .filter(Boolean)
+        .map((url) => {
+          try {
+            return new URL(url).hostname;
+          } catch (err) {
+            return;
+          }
+        })
+    );
+  }
+
   getInstrumentationId(): string {
     return 'http';
   }
 
   getInstrumentation(): HttpInstrumentation {
+    const ignoreConfig = this.getIgnoreConfig();
+
     return new HttpInstrumentation({
-      ignoreOutgoingUrls: this.getIgnoreConfig(),
-      ignoreIncomingPaths: this.getIgnoreConfig(),
+      ignoreOutgoingUrls: ignoreConfig,
+      ignoreIncomingPaths: [],
       requestHook: HttpHooks.requestHook,
       responseHook: HttpHooks.responseHook,
     });
   }
 
   private getIgnoreConfig(): IgnoreMatcher[] {
     return [
-      (url: string) =>
-        [
-          process.env.LUMIGO_ENDPOINT,
-          process.env.ECS_CONTAINER_METADATA_URI,
-          process.env.ECS_CONTAINER_METADATA_URI_V4,
-        ]
-          .filter(Boolean)
-          .some((v) => url.includes(v)),
+      (url: string) => this.ignoredHostnames.includes(new URL(url).hostname),
```

Comment:
don't we need the "try-catch" also here?

Reply:
yeah I think you are right

Reply:
Done in c4ae63c